boards/DIM/Inc/App/App_BoardStatusLeds.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include "App_BoardEnum.h"
+#include "App_SharedExitCode.h"
+
+/**
+ * Allocate and initialize a group of board status LEDs
+ * @param turn_on_dim_red A function that can be called to turn the DIM status
+ *                        LED red
+ * @param turn_on_dim_green A function that can be called to turn the DIM status
+ *                          LED green
+ * @param turn_on_dim_blue A function that can be called to turn the DIM status
+ *                         LED blue
+ * @param turn_on_dim_red A function that can be called to turn the DCM status
+ *                        LED red
+ * @param turn_on_dim_green A function that can be called to turn the DCM status
+ *                          LED green
+ * @param turn_on_dim_blue A function that can be called to turn the DCM status
+ *                         LED blue
+ * @param turn_on_dim_red A function that can be called to turn the FSM status
+ *                        LED red
+ * @param turn_on_dim_green A function that can be called to turn the FSM status
+ *                          LED green
+ * @param turn_on_dim_blue A function that can be called to turn the FSM status
+ *                         LED blue
+ * @param turn_on_dim_red A function that can be called to turn the BMS status
+ *                        LED red
+ * @param turn_on_dim_green A function that can be called to turn the BMS status
+ *                          LED green
+ * @param turn_on_dim_blue A function that can be called to turn the BMS status
+ *                         LED blue
+ * @param turn_on_dim_red A function that can be called to turn the PDM status
+ *                        LED red
+ * @param turn_on_dim_green A function that can be called to turn the PDM status
+ *                          LED green
+ * @param turn_on_dim_blue A function that can be called to turn the PDM status
+ *                         LED blue
+ * @return The created group of board status LEDs, whose ownership is given to
+ *         the caller
+ */
+struct BoardStatusLeds *App_BoardStatusLeds_Create(
+    void (*turn_on_dim_red)(void),
+    void (*turn_on_dim_green)(void),
+    void (*turn_on_dim_blue)(void),
+    void (*turn_on_dcm_red)(void),
+    void (*turn_on_dcm_green)(void),
+    void (*turn_on_dcm_blue)(void),
+    void (*turn_on_fsm_red)(void),
+    void (*turn_on_fsm_green)(void),
+    void (*turn_on_fsm_blue)(void),
+    void (*turn_on_bms_red)(void),
+    void (*turn_on_bms_green)(void),
+    void (*turn_on_bms_blue)(void),
+    void (*turn_on_pdm_red)(void),
+    void (*turn_on_pdm_green)(void),
+    void (*turn_on_pdm_blue)(void));
```

Comment:
Hmm, doesn't seem like this allows any way to turn off led's? Perhaps we have something like:
```
enum LED_STATE {ON, OFF};

void setDimRedLedState(LED_STATE state);
```
?

Reply:
So That's actually why I had initially but then I realized the spec never requires it to be in the OFF state. It seems like we only need RED, GREEN, or BLUE, but never OFF. That was the reasoning behind it. Does it seem reasonable to you?

Reply:
On a second though, let's just go with the suggestion here: https://github.com/UBCFormulaElectric/Consolidated-Firmware/pull/641#discussion_r433611469